assignment/index.md

```diff
@@ -14,3 +14,140 @@
 * Infinite scroll blog (async requests, DOM check scroll)
 * Basic animations
 * WebWorkers?
+
+# Build a SPA
+
+Inspiration:
+- https://web.archive.org/web/20140104191223/http://www.cse.unsw.edu.au:80/~cs2041/13s2/assignments/mekong/
+- Perhaps Andrew Taylor could provide the assignments from previous years?
+
+Build 2041StaGram. This assignment guides students to build an Instagram-like frontend to wrap a provided API. 
+
+The assignment intends to teach students to build a simple single page application which can fetch dynamic data from a HTTP API.
+
+Some of the skills/concepts this assignment aims to test (and build upon):
+
+- Simple event handling (buttons)
+- Advanced Mouse Events
+- Fetching data from an API
+- URL fragment based routing
+- Infinite scroll
+- CSS Animations
+- Web Workers
+  - Push Notifications
+  - Offline Support
+- Routing
+
+## API
+
+We can build API documentation using [Swagger](https://swagger.io/) or [Api Blueprint](https://apiblueprint.org/). A simple express JS API server can be built that is provided to students to allow them to develop offline. It could also be hosted somewhere to ease development when online? 
+
```

Comment:
Yes to all of this, as long we can get some time to build something. But it shouldn't be too hard.